### PR TITLE
Block breaking animations

### PR DESCRIPTION
**Pull request by endevrr**
Completely accurate animations other than if the player is in under water or if the players feet are not touching the ground, this is due to Geyser not having access to the server api

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/entity/PlayerEntity.java [hunk 1]
```diff
@@ -50,13 +51,16 @@
     private String username;
     private long lastSkinUpdate = -1;
     private boolean playerList = true;
+    private final EntityEffectCache effectCache;
+
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/PlayerEntity.java]: Random space here not needed.

File: connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerActionAckTranslator.java [hunk 2]
```diff
@@ -38,6 +48,45 @@ public void translate(ServerPlayerActionAckPacket packet, GeyserSession session)
             case FINISH_DIGGING:
                 ChunkUtils.updateBlock(session, packet.getNewState(), packet.getPosition());
                 break;
+
+            case START_DIGGING: {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerActionAckTranslator.java]: Opening and closing brackets here are not needed.

File: connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerActionAckTranslator.java [hunk 3]
```diff
@@ -38,6 +48,45 @@ public void translate(ServerPlayerActionAckPacket packet, GeyserSession session)
...
+                    nbtData = item.getNbt();
+                }
+                double breakTime = Math.ceil(BlockUtils.getBreakTime(blockHardness, packet.getNewState().getId(), itemEntry, nbtData, session.getPlayerEntity()) * 20);
+                int data = (int) (65535 / breakTime);
+                levelEvent.setData((int) (65535 / breakTime));
+                session.getUpstream().sendPacket(levelEvent);
+                break;
+            }
+
+            case CANCEL_DIGGING: {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerActionAckTranslator.java]: Same as above.

**Comment by RednedEpic** [comment]
Is it possible to rebase this on top of master?

**Comment by endevrr** [comment]
Master doesn't have the inventory code required for the tool logic (most of the logic)

**Pull request merged**